src/library/autodj/dlgautodj.cpp

```diff
@@ -126,6 +150,17 @@ DlgAutoDJ::~DlgAutoDJ() {
     delete m_pTrackTableView;
 }
 
+void DlgAutoDJ::setupActionButton(QPushButton* pButton, void (DlgAutoDJ::*pSlot)(bool),
+                           QString skinButtonName, QString fallbackText) {
+    connect(pButton, &QPushButton::clicked, this, pSlot);
+    QString iconPath = m_icons.value(skinButtonName).states.value("Off").pixmapSource.getPath();
+    if (iconPath.isEmpty()) {
+        pButton->setText(fallbackText);
+    } else {
+        pButton->setIcon(QIcon(iconPath));
```

Comment:
For some reason this makes the icons always have a height of 16px in the skin, regardless of the actual icon size.
All Library QPushButtons have IDs, and setting the size (min-width and height) as well as the icons for `!checked` and `checked` state in style.qss works reliably. Tested on linux with 100% and 200% scaling.
I have no idea how to fix the lines above to make the icons scale with the button, so can we remove the styling from C++ code and do it all in qss?
Also the toolpis and the icons are descriptive, why do we need the fallback texts?

Reply:
>  Also the toolpis and the icons are descriptive, why do we need the fallback texts?

The fallback text is needed for skins that do not specify the icons, which is currently all skins for most of the buttons. This will also preserve backwards compatibility with skins made for earlier versions of Mixxx.

Reply:
okay, I see.
But unless you have a fix handy to solve the icon size issue described above I'll workaround it by using a blank dummy icon to hide the text and do all styling in the style sheets per skin.
This works for all current official skins and preserves the skin backwards compatibility, and it's also where margins, padding, bg colors and border images are defined.

Reply:
dummy icons per skin (transparent 1x1px) hide the text and allow to set the icons properly sized via qss.
I don't see another way, and working with 16x16px icons  via <Button> nodes is pointless IMO

Reply:
Is this issue solved now? Was it a regression from this PR?

Reply:
It's not a regression; the functionality didn't exist before this PR.

Reply:
The Icon looks OK on my system. When does this issue appear?

Reply:
Okay, I understand the issue now. QIcons are not simply QPixmaps that can be set as icons; QIcons can be shown at various sizes. Changing the icon size that a button displays requires calling QAbstractButton::setIconSize. However, I wouldn't want to hardcode one size for every skin, nor require the image files to be certain dimensions. I think it would be best to add an option to hide the text and let the skins set icons in QSS.